opm/input/eclipse/Schedule/Well/GridIndependentWellKeywordHandlers.cpp

```diff
@@ -0,0 +1,243 @@
+                    wells.insert(name);
+                }
+
+                if (connections->empty() && well2.getConnections().empty()) {
+                    const auto& location = handlerContext.keyword.location();
+                    const auto msg = fmt::format(R"(Problem with COMPTRAJ/{}
+In {} line {}
+Well {} is not connected to grid - will remain SHUT)",
+                                                 name,
+                                                 location.filename,
+                                                 location.lineno,
+                                                 name);
+                    OpmLog::warning(msg);
+                }
+
+                double diameter = record.getItem("DIAMETER").getSIDouble(0);
+                process_segments(handlerContext, name, intersections, wellPathGeometry, diameter);
+
+                handlerContext.state().wellgroup_events().addEvent(
+                    name, ScheduleEvents::COMPLETION_CHANGE);
+            }
+        }
+
+        handlerContext.state().events().addEvent(ScheduleEvents::COMPLETION_CHANGE);
+
+        // In the case the wells reference depth has been defaulted in the
+        // WELSPECS keyword we need to force a calculation of the wells
+        // reference depth exactly when the COMPTRAJ keyword has been
+        // completely processed.
+        for (const auto& wname : wells) {
+            auto well = handlerContext.state().wells.get(wname);
+            well.updateRefDepth();
+
+            handlerContext.state().wells.update(std::move(well));
+            handlerContext.comptraj_handled(wname);
+        }
+
+        if (!wells.empty()) {
+            handlerContext.record_well_structure_change();
+        }
+    }
+
+
+    void handleWELTRAJ(HandlerContext& handlerContext)
+    {
+        for (const auto& record : handlerContext.keyword) {
+            const std::string& wellNamePattern = record.getItem("WELL").getTrimmedString(0);
+            const auto wellnames = handlerContext.wellNames(wellNamePattern, false);
+
+            for (const auto& name : wellnames) {
+                auto well2 = handlerContext.state().wells.get(name);
+                auto connections
+                    = std::make_shared<WellConnections>(WellConnections(well2.getConnections()));
+                connections->loadWELTRAJ(
+                    record, handlerContext.grid, name, handlerContext.keyword.location());
+                if (well2.updateConnections(connections, handlerContext.grid)) {
+                    handlerContext.state().wells.update(well2);
+                    handlerContext.record_well_structure_change();
+                }
+                handlerContext.state().wellgroup_events().addEvent(
+                    name, ScheduleEvents::COMPLETION_CHANGE);
+                const auto& md = connections->getMD();
+                if (!std::is_sorted(std::begin(md), std::end(md))) {
```

Comment:
std::is_sorted allows equal elements. it is different from the `strictly increasing` message.  Do we allow equal elements here?